sdk-src/src/main/java/com/hpe/adm/nga/sdk/network/google/GoogleHttpClient.java

```diff
@@ -255,22 +255,49 @@ private OctaneHttpResponse execute(OctaneHttpRequest octaneHttpRequest, int retr
 
     private HttpResponse executeRequest(final HttpRequest httpRequest) throws IOException {
         logger.debug(LOGGER_REQUEST_FORMAT, httpRequest.getRequestMethod(), httpRequest.getUrl().toString(), httpRequest.getHeaders().toString());
-        final ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+
         final HttpContent content = httpRequest.getContent();
-        if (content != null) {
-            content.writeTo(byteArrayOutputStream);
 
-            //Do not print the body of the sign in request
-            if (!httpRequest.getUrl().toString().contains(OAUTH_AUTH_URL)) {
-                logger.debug("Content: " + byteArrayOutputStream.toString());
-            }
+        // Make sure you don't log any http content send to the login rest api, since you don't want credentials in the logs
+        if (content != null && logger.isDebugEnabled() && !httpRequest.getUrl().toString().contains(OAUTH_AUTH_URL)) {
+            logHttpContent(Level.DEBUG, content);
         }
 
         HttpResponse response = httpRequest.execute();
         logger.debug(LOGGER_RESPONSE_FORMAT, response.getStatusCode(), response.getStatusMessage(), response.getHeaders().toString());
         return response;
     }
 
+    /**
```

Comment:
I'm guessing you meant this for the line above, with the logging. If you log the http content object, the input streams inside of it will be closed. Also if your uploading a file, you don't really want to log 100mb of useless bytes. So Based on the request/response http content type, I decide wheter or not to debug log it's content